python/ray/autoscaler/_private/local/node_provider.py

```diff
@@ -196,7 +207,14 @@ def node_tags(self, node_id):
         return self.state.get()[node_id]["tags"]
 
     def external_ip(self, node_id):
-        return socket.gethostbyname(node_id)
+        node_state = self.state.get()[node_id]
+        ext_ip = node_state.get("external_ip")
+        # If user has supplied an external ip, return that.
+        # (Useful when calling ray up from outside the network.)
```

Comment:
I m not sure why we need this external_ip...

Reply:
I need it to run ray up from my laptop.